packages/vue-app/template/client.js

```diff
@@ -146,9 +146,12 @@ async function loadAsyncComponents(to, from, next) {
       // Add a marker on each component that it needs to refresh or not
       const startLoader = Components.some((Component) => {
         const watchQuery = Component.options.watchQuery
-        if (watchQuery === true) return true
-        if (Array.isArray(watchQuery)) {
+        if (watchQuery === true) {
+          return true
+        } else if (Array.isArray(watchQuery)) {
           return watchQuery.some(key => this._diffQuery[key])
+        } else if (typeof watchQuery === 'function') {
+          return watchQuery(from.query, to.query)
```

Comment:
Maybe reverse the arguments as well? Is more in line with a Vue watcher which receives `newVal, oldVal`

Reply:
We can also use named params :) `{ from, to }`

Reply:
Should be ok as well indeed.

Do you know if (at least in v8) thats optimized exactly the same as using arguments? Eg i know v8 optimizes calls to functions which have destructed args, but does it also optimize that object fully away if the receiving arg only destructures into one property? Sounds logical to do that if you are already optimizing destructures, but dont know for sure

Reply:
Suggest to read about inline caches: https://mathiasbynens.be/notes/shapes-ics

Reply:
Thanks, interesting read. This article answered my question a bit more: https://codeburst.io/es6s-function-destructuring-assignment-is-not-free-lunch-19caacc18137

Reply:
Babel makes it different: 

https://babeljs.io/en/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYewdgzgLgBAZmGBeGAKA3vATiAtgGhihBgF8BKZAPhnVKA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.5.5&externalPlugins=